### DISCUSSION TITLE
PDF Download Does't work anymore

### DISCUSSION THREAD

**Discussion by MarcRevo** [open-webui/open-webui#5222] [General]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:**  v0.3.19
- **Ollama (if applicable):** 0.3.9

- **Operating System:** Ubuntu 20.04
- **Browser (if applicable):** Chrome

**Confirmation:**

- [ x ] I have read and followed all the instructions provided in the README.md.
- [ x ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ x ] I have included the Docker container logs.
- "POST /api/v1/utils/pdf HTTP/1.1" 500 Internal Server Error
- [ x ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Go to your chat, then click download PDF.

## Actual Behavior:

Does not start the download. And in the docker log there will be a 500 internal error

**Docker Container Logs:**
"POST /api/v1/utils/pdf HTTP/1.1" 500 Internal Server Error

**Comment by silentoplayz** [comment]
I've tested each `Download` option of a chat on the latest dev commit and I can confirm that the `PDF document (.pdf)` option, when clicked, does not appear to download the chat as a PDF file.

Confirming that this is indeed a bug that would need fixed.

**Comment by silentoplayz** [comment]
Here's a copy/paste of my Docker logs generated from pressing on the `PDF document (.pdf)` option to download a chat within Open WebUI (logs were cleared directly before pressing on the button): [Docker_logs.txt](https://github.com/user-attachments/files/16908705/Docker_logs.txt)

**Comment by tjbck** [comment]
Should be fixed on dev!

**Reply by Tianyusz** [reply]
This issue is not fixed yet

**Reply by silentoplayz** [reply]
> This issue is not fixed yet

What's not working for you exactly? More elaboration or details would be helpful here. Personally, I've tested downloading a chat as a .PDF file on the dev branch and it downloaded successfully on the latest version of both Firefox and Chrome browsers.

**Reply by socketbox** [reply]
The PDF that's generated is blank. It has no content. I've tried both light and dark themes, thinking that operating in the latter was the issue.

**Comment by aario** [comment]
Mine (v0.4.5) gives the error:
```Character "’" at index 125 in text is outside the range of characters supported by the font used: "courier". Please consider using a Unicode font.```
When I try to download a conversation containing german special characters like `Ü` or `Ö`.

**Comment by iunknown80** [comment]
I am facing this error "Character "😊" at index 55 in text is outside the range of characters supported by the font used: "courier". Please consider using a Unicode font." when I try to download a chat as pdf. 
This can be easily reproduced by starting a new chat. Just say "HI" and then try to download the chat conversation as a PDF. There is no issue with text files or JSON. 
Version 4.8

**Comment by iunknown80** [comment]
Here is the link to the changes if you stumble on this issue with 0.4.8. 

https://github.com/open-webui/open-webui/commit/8dcee6b6ed4833d58e521b15a7f2ec93464cc45e

**Comment by socketbox** [comment]
Why was the Issue (#5221) for this bug closed and moved to a discussion? I'm on v0.7.2 and this is still broken.

**Reply by Classic298** [reply]
Because it is insanely hard to fix and has been postponed to a future version, though i think @ShirasawaSama is working on a fix here.

**Reply by ShirasawaSama** [reply]
I've been working on fixing this issue recently and will submit a pull request soon. Due to the large amount of code involved, it's still under development at this time.

**Reply by socketbox** [reply]
Update: sorry, @ShirasawaSama , I was in the process of responding to the first comment when you posted. Thank you for working on this. Please let me know if I can help in any way.

Thanks, @Classic298 . Is there a way that I can provide debugging information? I'm running Open WebUI in a docker compose environment and have set `GLOBAL_LOGGING_LEVEL=DEBUG` in `env:` and I still only see `INFO` in the console, even when I download a PDF (which I've attached).
[openwebui_kimi-2.5_voting.pdf](https://github.com/user-attachments/files/25271191/openwebui_kimi-2.5_voting.pdf)


```
[2026-02-12 14:21:05]» pdfinfo ~/Downloads/pdf/openwebui_kimi-2.5_voting.pdf
Producer:        jsPDF 4.0.0
CreationDate:    Thu Feb 12 14:16:04 2026 EST
Custom Metadata: no
Metadata Stream: no
Tagged:          no
UserProperties:  no
Suspects:        no
Form:            none
JavaScript:      no
Pages:           32
Encrypted:       no
Page size:       595.28 x 841.89 pts (A4)
Page rot:        0
File size:       42160 bytes
Optimized:       no
PDF version:     1.3
```

Am I misinterpreting this? It seems like the code is creating an enormous blank raster image for each page...
```
[2026-02-12 14:22:05]» pdfimages -list ~/Downloads/pdf/openwebui_kimi-2.5_voting.pdf
page   num  type   width height color comp bpc  enc interp  object ID x-ppi y-ppi size ratio
--------------------------------------------------------------------------------------------
   1     0 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   2     1 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   3     2 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   4     3 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   5     4 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   6     5 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   7     6 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   8     7 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
   9     8 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  10     9 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  11    10 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  12    11 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  13    12 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  14    13 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  15    14 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  16    15 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  17    16 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  18    17 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  19    18 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  20    19 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  21    20 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  22    21 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  23    22 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  24    23 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  25    24 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  26    25 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  27    26 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  28    27 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  29    28 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  30    29 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  31    30 image    1600  2262  rgb     3   8  jpeg   no        81  0   194   194 21.5K 0.2%
  32    31 image    1600   282  rgb     3   8  jpeg   no        82  0   194   194 3457B 0.3%
```

**Reply by ShirasawaSama** [reply]
You can export your chat data as a JSON file and upload it to Google Drive or another location. I'll download it and test it with the code I've fixed so far. I'm currently in need of more data. Thanks.

**Reply by ShirasawaSama** [reply]
Could you please help me test the PDF export?

**Comment by ShirasawaSama** [comment]
https://github.com/open-webui/open-webui/pull/21364
